Foundation/NSUserDefaults.swift

```diff
@@ -7,46 +7,286 @@
 public let NSArgumentDomain: String = "NSArgumentDomain"
 public let NSRegistrationDomain: String = "NSRegistrationDomain"
 
+private var registeredDefaults = [String: AnyObject]()
+private var sharedDefaults = NSUserDefaults()
+
 public class NSUserDefaults : NSObject {
+    private let suite: String?
+    
+    public class func standardUserDefaults() -> NSUserDefaults {
+        return sharedDefaults
+    }
     
-    public class func standardUserDefaults() -> NSUserDefaults { NSUnimplemented() }
-    public class func resetStandardUserDefaults() { NSUnimplemented() }
+    public class func resetStandardUserDefaults() {
+        sharedDefaults.synchronize()
+        sharedDefaults = NSUserDefaults()
+    }
     
-    public convenience override init() { NSUnimplemented() }
-    public init?(suiteName suitename: String?)  { NSUnimplemented() } //nil suite means use the default search list that +standardUserDefaults uses
+    public convenience override init() {
+        self.init(suiteName: nil)!
+    }
     
-    public func objectForKey(defaultName: String) -> AnyObject? { NSUnimplemented() }
-    public func setObject(value: AnyObject?, forKey defaultName: String) { NSUnimplemented() }
-    public func removeObjectForKey(defaultName: String) { NSUnimplemented() }
+    /// nil suite means use the default search list that +standardUserDefaults uses
+    public init?(suiteName suitename: String?) {
+        suite = suitename
+    }
+    
+    public func objectForKey(defaultName: String) -> AnyObject? {
+        func getFromRegistered() -> AnyObject? {
+            return registeredDefaults[defaultName]
+        }
+        
+        guard let anObj = CFPreferencesCopyAppValue(defaultName._cfObject, suite?._cfObject ?? kCFPreferencesCurrentApplication) else {
+            return getFromRegistered()
+        }
+        
+        //Force the returned value to an NSObject
+        switch CFGetTypeID(anObj) {
+        case CFStringGetTypeID():
+            return (anObj as! CFStringRef)._nsObject
+            
+        case CFNumberGetTypeID():
+            return (anObj as! CFNumberRef)._nsObject
+            
+        case CFURLGetTypeID():
+            return (anObj as! CFURLRef)._nsObject
+            
+        case CFArrayGetTypeID():
+            return (anObj as! CFArrayRef)._nsObject
+            
+        case CFDictionaryGetTypeID():
+            return (anObj as! CFDictionaryRef)._nsObject
+
+        case CFDataGetTypeID():
+            return (anObj as! CFDataRef)._nsObject
+            
+        default:
+            return getFromRegistered()
+        }
+    }
+    public func setObject(value: AnyObject?, forKey defaultName: String) {
+        guard let value = value else {
+            CFPreferencesSetAppValue(defaultName._cfObject, nil, suite?._cfObject ?? kCFPreferencesCurrentApplication)
+            return
+        }
+        
+        var cfType: CFTypeRef? = nil
+		
+		//FIXME: is this needed? Am I overcomplicating things?
```

Comment:
Yea this is probably not needed at all since all of these types are already bridged to CF